### ISSUE TITLE
[Packager] Source Map not correctly referenced from JS bundle.

### ISSUE THREAD

**Issue by qbig** [facebook/react-native#5277]
If I do the following bundle command, specifying the sourcemap file to be "index.android.map"
```
react-native bundle --platform android --dev false --entry-file index.android.js   --bundle-output android/app/src/main/assets/index.android.bundle   --assets-dest android/app/src/main/res/ --sourcemap-output android/app/src/main/assets/index.android.map
```
but in the output bundle file, 'bundle.js‘ is used as the sourceMappingURL
```
//# sourceMappingURL=bundle.js
```
If sourceMappingURL is the reference to the sourcemap file, shouldn't be "index.android.map" instead ? Or is 'bundle.js' use somewhere else internally?

**Comment by skevy** [comment]
@qbig can you try this with master and let me know the results?

**Comment by qbig** [comment]
I tried this, where "/Users/qiaoliang89/Downloads/myRn/react-native" is my clone from the repo.
```
node /Users/qiaoliang89/Downloads/myRn/react-native/local-cli/cli.js  bundle --platform android --dev false --entry-file index.android.js   --bundle-output index.android.bundle   --assets-dest /android/app/src/main/res/ --sourcemap-output index.android.map
```
but got 
```
/Users/qiaoliang89/Downloads/myRn/react-native/node_modules/babel-core/lib/transformation/file/options/option-manager.js:181
          throw new ReferenceError(messages.get("pluginUnknown", plugin, loc, i));
          ^

ReferenceError: Unknown plugin "syntax-async-functions" specified in "foreign" at 0
    at /Users/qiaoliang89/Downloads/myRn/react-native/node_modules/babel-core/lib/transformation/file/options/option-manager.js:181:17
    at Array.map (native)
    at Function.normalisePlugins (/Users/qiaoliang89/Downloads/myRn/react-native/node_modules/babel-core/lib/transformation/file/options/option-manager.js:163:20)
    at OptionManager.mergeOptions (/Users/qiaoliang89/Downloads/myRn/react-native/node_modules/babel-core/lib/transformation/file/options/option-manager.js:259:36)
    at compile (/Users/qiaoliang89/Downloads/myRn/react-native/node_modules/babel-register/lib/node.js:87:15)
    at loader (/Users/qiaoliang89/Downloads/myRn/react-native/node_modules/babel-register/lib/node.js:128:14)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/qiaoliang89/Downloads/myRn/react-native/node_modules/babel-register/lib/node.js:138:7)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
```

might be #5135  ?

**Comment by qbig** [comment]
fixed with  #5288
